cmd/archeio/docs/testing.md

```diff
@@ -12,7 +12,7 @@ methods, we also have unit tests covering the HTTP Handlers and full
 
 **This level of coverage must be maintained**, it is imperative that we have robust
 testing in this project that may soon serve all Kubernetes project image downloads.
-TODO: this should be enforced by CI. Currently it is enforced by reviewers.
+We automatically enforce 100% code coverage for archeio sources.
```

Comment:
is this for real?

Reply:
yes*, after this PR :-)

test test presubmit runs this tool now https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/registry.k8s.io/179/pull-oci-proxy-test/1635909009225551872

\* not on main.go ... yet. I need to rework the integration tests. that's another PR, prioritizing this one. also not zz_generated* which is filtered from all.cov => filtered.cov, but I think we can remove filtering that and we definitely cover it actually.

we're excluding other sources currently, such as the cmd/geranos sources, and some of the ip ranges code generator

Reply:
... and I just added a commit to stop filtering out generated files. so just cmd/archeio/main.go needs fixing.

Reply:
I always found 100% excessive, my observation is that around 80 or 85 is a good coverage level

Reply:
That might generally be true but I don't think so for a small stateless HTTP redirector.

we can and absolutely should test all code paths. There's a high risk to production workloads if we have any bugs and the service logic has minimal external dependencies that are easily mocked (HEAD requests to object storage)

some of the coverage is more integration test than unit but go doesn't distinguish and  happy to count both as coverage

Reply:
We're not requiring any particular level currently for the "ignored" files which are used for other less critical tools.

it's also ironically harder to effectively e2e test all codepaths in the registry server than unit / integration because we route based on client address so ...